frontend/src/views/new-project/NewProjectView.tsx

```diff
@@ -60,66 +60,101 @@ const createProjectMutation = gql`
  */
 export const NewProjectView = () => {
   const [createProject, { loading, data, error }] = useMutation(createProjectMutation);
-  const [state, dispatch] = useReducer(reducer, initialState);
-
-  const { viewState, name, nameMessage, nameMessageSeverity, message, newProjectId } = state;
+  const [{ value, context }, dispatch] = useMachine<NewProjectViewContext, NewProjectEvent>(newProjectViewMachine);
+  const { newProjectView, toast } = value as SchemaValue;
+  const { name, newProjectId, message } = context;
 
   const onNameChange = (event) => {
-    const name = event.target.value;
-    dispatch({ type: HANDLE_CHANGE_NAME__ACTION, name });
+    const value = event.target.value;
+    const changeNameEvent: ChangeNameEvent = { type: 'CHANGE_NAME', name: value };
+    dispatch(changeNameEvent);
   };
 
   const onCreateNewProject = async (event) => {
     event.preventDefault();
-
     const variables = {
       input: {
         name: name.trim(),
         visibility: 'PUBLIC',
       },
     };
+    const submitEvent: SubmitEvent = { type: 'SUBMIT' };
+    dispatch(submitEvent);
     createProject({ variables });
   };
 
   useEffect(() => {
     if (!loading && !error && data) {
       if (error) {
-        dispatch({ type: HANDLE_SUBMIT_FAILURE__ACTION, reponse: error });
+        let errorMessage = error.message;
+        const showToastEvent: ShowToastEvent = { type: 'SHOW_TOAST', message: errorMessage };
+        dispatch(showToastEvent);
       } else if (data) {
-        dispatch({ type: HANDLE_SUBMIT__ACTION, response: data });
+        let id = data.createProject?.project?.id;
+        if (id) {
```

Comment:
You may want to discriminate on the type of the response instead of the presence of one field. Something like that:

const typename = data.createProject.__typename;
if (typename === 'CreateProjectSuccessPayload') {
  const { id } = data.createProject.project;
} else if (typename === 'ErrorPayload') {
  const { message } = data.createProject;
}
```

Inside of the first if, you know for a fact that project and id cannot be null and inside of the second one message cannot be null either. That's part of the GraphQL API, the server will make sure of it, otherwise you will receive an error.